mysqld_exporter.go

```diff
@@ -124,40 +125,66 @@ func (e *Exporter) scrape(scrapes chan<- []string) {
 	// fetch slave status
 	rows, err = db.Query("SHOW SLAVE STATUS")
 	if err != nil {
-		log.Println("error running show slave query on database: ", err)
+		log.Println("error running show slave query on database:", err)
 		e.error.Set(1)
 		e.duration.Set(float64(time.Now().UnixNano()-now) / 1000000000)
 		return
 	}
 	defer rows.Close()
 
+	var slaveCols []string
+
+	slaveCols, err = rows.Columns()
+	if err != nil {
+		log.Println("error retrieving column list:", err)
+		e.error.Set(1)
+		e.duration.Set(float64(time.Now().UnixNano()-now) / 1000000000)
+		return
+	}
+
+	var slaveData = make([]sql.RawBytes, len(slaveCols))
+
+	scanArgs := make([]interface{}, len(slaveCols))
+	for i := range slaveData {
+		scanArgs[i] = &slaveData[i]
+	}
+
 	for rows.Next() {
-		// get RawBytes from data
-		err = rows.Scan(&key, &val)
+
+		err = rows.Scan(scanArgs...)
 		if err != nil {
-			log.Printf("error getting result set: ", err)
+			log.Println("error retrieving result set:", err)
+			e.error.Set(1)
+			e.duration.Set(float64(time.Now().UnixNano()-now) / 1000000000)
 			return
 		}
 
-		var res []string = make([]string, 2)
-		res[0] = string(key)
-		if string(key) == "Slave_IO_State" {
-			if strings.HasPrefix(string(val), "Waiting") {
+	}
+
+	for i, col := range slaveCols {
+
+		if col == "Slave_IO_Running" || col == "Slave_SQL_Running" {
+
+			var res []string = make([]string, 2)
+			res[0] = col
+
+			if string(slaveData[i]) == "Yes" {
 				res[1] = "1"
 			} else {
 				res[1] = "0"
 			}
-		} else {
-			if string(val) == "Yes" {
-				res[1] = "1"
-			} else if string(val) == "No" {
-				res[1] = "0"
-			} else {
-				res[1] = string(val)
-			}
+
+			scrapes <- res
 		}
 
-		scrapes <- res
+		if col == "Seconds_Behind_Master" {
```

Comment:
I skipped the others because they're generally either junk, strings that have no sensible integer conversion, or both.  I can put them in if you'd prefer, but they're almost entirely useless except for these three.